scvi/dataset/utils.py

```diff
@@ -47,40 +50,137 @@ def get_indices(labels, n_labelled_samples_per_class_array):
     return permutation_idx[indices[:total_labelled]], permutation_idx[indices[total_labelled:]]
 
 
-def get_data_loaders(gene_dataset, n_labelled_samples_per_class, **kwargs_data_loader):
-    """
-    For use in train_semi_supervised
-    Get labelled/unlabelled/all data_loaders from gene_dataset
-    :param gene_dataset: the dataset to consider
-    :param n_labelled_samples_per_class: number of labelled samples per class
-    :param kwargs_data_loader: any additional keyword arguments to pass to the dataloaders
-    :return: data_loader_labelled/data_loader_unlabelled/data_loader_all
-    """
-    indices_labelled, indices_unlabelled = (
-        get_indices(gene_dataset.labels, [n_labelled_samples_per_class] * gene_dataset.n_labels)
-    )
-    data_loader_all = DataLoader(gene_dataset, collate_fn=gene_dataset.collate_fn, shuffle=True, **kwargs_data_loader)
-    data_loader_labelled = DataLoader(gene_dataset, collate_fn=gene_dataset.collate_fn,
-                                      sampler=SubsetRandomSampler(indices_labelled), **kwargs_data_loader)
-    data_loader_unlabelled = DataLoader(gene_dataset, collate_fn=gene_dataset.collate_fn,
-                                        sampler=SubsetRandomSampler(indices_unlabelled), **kwargs_data_loader)
-    return data_loader_all, data_loader_labelled, data_loader_unlabelled
-
-
-def get_raw_data(*data_loaders):
-    """
-    From a list of data_loaders, return the numpy array suitable for classification with sklearn
-    :param data_loaders: a sequence of data_loaders arguments
-    :return: a sequence of (data, labels) for each data_loader
-    """
+def zip_first_cycle(*args):
+    return zip(args[0], *[cycle(arg) for arg in args[1:]])
+
+
+class DataLoaders:
```

Comment:
Oh, I guess `zip_first_cycle` could be moved to the `DataLoaders` class. Actually, if you just call `zip_first_cycle` once, since it's a one line function, you might just get rid of it...call `zip(...)` directly in `__iter__`.

Reply:
#95 moves `zip_first_cycle` and renames `utils.py` to `data_loaders.py`.

`get_indices` is used in :
```python
class SemiSupervisedDataLoaders(DataLoaders):
    to_monitor = ['labelled', 'unlabelled']

    def __init__(self, gene_dataset, n_labelled_samples_per_class=50, seed=0, **data_loaders_kwargs):
        """
        :param n_labelled_samples_per_class: number of labelled samples per class
        """
        super(SemiSupervisedDataLoaders, self).__init__(gene_dataset, **data_loaders_kwargs)
        indices_labelled, indices_unlabelled = (
            get_indices(gene_dataset.labels, [n_labelled_samples_per_class] * gene_dataset.n_labels, seed=seed)
        )
```
To get labelled/unlabelled split. Should I move the function inside the \_\_init\_\_, since it is only used here ?

Reply:
Sure, that'd be great!

Reply:
ok that's done ! I squash&merge